### ISSUE TITLE
Cannot use the builder/react sdk in the new version of shopify-hydrogen using Remix

### ISSUE THREAD

**Issue by tylerscave** [BuilderIO/builder-shopify-hydrogen#12]
Shopify-Hydrogen has recently released a new version using Remix (https://github.com/Shopify/hydrogen/tree/2023-01/templates/demo-store)

The following error is generated when attempting to use anything from the `@builder.io/react` sdk in any of the routes. The only place I seem to be able to use `@builder.io/react` successfully is in the `entry.client.tsx` file. 
![evalError](https://user-images.githubusercontent.com/14061153/218607234-491373ae-0a86-4ada-a949-621d8209370c.png)

Our project was created using the newest template from Shopify-Hydrogen.  In my case, this error is generated when using `useIsPreviewing()` or the `BuilderComponent`

As an example, the error is generated in the component shown below:

```typescript
import {BuilderComponent, useIsPreviewing} from '@builder.io/react';
import {BuilderContent} from '@builder.io/sdk';
import {Suspense} from 'react';
import tw from 'tailwind-styled-components';
import {NotFound} from '../NotFound';

export const PaxBuilderPage = ({
  builderContent,
  builderModel,
}: {
  builderContent: Omit<BuilderContent, 'variations' | 'data'>;
  builderModel: string;
}) => {
  const isPreviewingInBuilder = useIsPreviewing();

  if (!builderContent && !isPreviewingInBuilder) {
    return <NotFound />;
  }

  return (
    <Suspense>
      <BuilderContentContainer>
        <BuilderComponent content={builderContent} model={builderModel} />
      </BuilderContentContainer>
    </Suspense>
  );
};

const BuilderContentContainer = tw.div`
  flex
  h-fit
  flex-col
`;
```

The above `PaxBuilderPage` component is being used in our site's `Homepage` (route = '/') as shown below. That said, the same error will still occur when the `BuilderComponent` is imported  from `@builder.io/react` directly into the `Homepage` 
```typescript
import {builder, BuilderContent} from '@builder.io/sdk';
import {Await, useLoaderData} from '@remix-run/react';
import builderConfig from '../../../builderConfig.json';
import {PaxBuilderPage} from '../../components/builder/PaxBuilderPage';

builder.init(builderConfig.apiKey);

const BUILDER_MODEL = 'page';

export async function loader() {
  const builderContent = await builder
    .get(BUILDER_MODEL, {
      userAttributes: {
        urlPath: '/',
      },
    })
    .toPromise();

  if (!builderContent) {
    throw new Response('Not found', {status: 404});
  }

  return builderContent;
}

// this gives full compatibility to BuilderContent type and Remix starter
// See: https://github.com/BuilderIO/builder/issues/1387#issuecomment-1397442797
type BuilderContentRemix = Omit<BuilderContent, 'variations' | 'data'>;

export default function Homepage() {
  const builderContent: BuilderContentRemix =
    useLoaderData<BuilderContentRemix>();

  return (
    <Await resolve={builderContent}>
      <PaxBuilderPage
        builderContent={builderContent}
        builderModel={BUILDER_MODEL}
      />
    </Await>
  );
}
```

**Comment by mrkoreye** [comment]
Hi @tylerscave thank you for the detailed issue! We [fixed a very similar issue](https://github.com/BuilderIO/builder/pull/1311) last October, but it looks like with the new Hydrogen version it is back 😢 . We will investigate and report back when we have a fix.

**Comment by alejuu** [comment]
Hi, I accidentally created another similar case that I already closed. I'm having a similar issue, I'm trying to add the Builder SDK similarly as added in this repository, but it isn't working. I'm not getting any errors on the console.

**Comment by terza98** [comment]
Hey guys - any updates on this fix ? We're having the same issue

**Comment by akshatgarg12** [comment]
Hey @terza98 still looking into this, will update as soon as we get the solution

**Comment by terza98** [comment]
Hey @akshatgarg12 - do you have any ETA on when this might be resolved?

**Comment by terza98** [comment]
Awesome ! thanks for fixing it so fast

**Comment by teleaziz** [comment]
Fix is released on @builder.io/react@2.0.16, thanks for reporting this issue

**Comment by tylerscave** [comment]
Thanks for getting this out! We are finally unblocked, and everything seems to mostly be working now! 🎉

One issue that I noticed, and I'm not totally sure yet if its related to this package, but `BuilderComponent`s (ie Builder pages) load excruciatingly slowly when running locally (which affects how we use the builder content editor)... They are quick and snappy in our deployed code tho 🤔

**Comment by terza98** [comment]
@tylerscave Same issue here

**Comment by mrkoreye** [comment]
@tylerscave sorry to hear about the slow loading locally. Can you open a new issue with reproduction steps so we can take a look?
